test/OpenTelemetry.Tests/Impl/Trace/Export/BatchingSpanProcessorTests.cs

```diff
@@ -268,7 +268,7 @@ public async Task ShutdownOnNotEmptyQueueFullFlush()
                     await spanProcessor.ShutdownAsync(cts.Token);
                 }
 
-                Assert.Equal(spans.Count, spanExporter.ExportedSpans.Length);
+                Assert.Equal(0, spanExporter.ExportedSpans.Length);
```

Comment:
I don't understand. After you've shut down the processor, all spans should be exported. Why 0?

Reply:
This is the code in `ShutdownAsync` that has not been called before:

```csharp
public override Task ShutdownAsync(CancellationToken cancellationToken)
{
  while (spanDataList.TryDequeue(out var _))
   {
   }

  return Task.CompletedTask;
}
```

Now that it gets called, the `spanDataList` collection (backing field for `ExportedSpans`) gets depleted. Hence, expected size is 0.

Reply:
Oh, i see. I believe the intent of the test was to make sure all spans are exported. Now it's not possible to tell if they were (or nothing ever was exported). can we reflect both in tests: that spans were exported and shutdown was called?

Reply:
I removed the `TryDequeue` loop since this behavior isn't tested for anyway and test for the expected call to `ShutDownAsync` explicitly now.